lib/std/crypto/aes_cmac.zig

```diff
@@ -0,0 +1,171 @@
+/// RFC 4493 - The AES-CMAC Algorithm
+/// https://www.rfc-editor.org/rfc/rfc4493
+const std = @import("std");
+const Aes128 = std.crypto.core.aes.Aes128;
+const print = std.debug.print;
+
+pub fn AesCmac() type {
+    return struct {
+        const Self = @This();
+        var k1: [16]u8 = undefined;
+        var k2: [16]u8 = undefined;
+        var ctx: std.crypto.core.aes.AesEncryptCtx(Aes128) = undefined;
+
+        /// Create a new cmac context with the given key.
+        pub fn init(key: [16]u8) Self {
+            ctx = Aes128.initEnc(key);
+            k1 = [_]u8{0} ** 16;
+            k2 = [_]u8{0} ** 16;
+            ctx.encrypt(&k1, &k1);
+            makeKn(&k1, &k1);
+            makeKn(&k2, &k1);
+            return Self{};
```

Comment:
Looks like there's something wrong here. Keys are computed, but what is returned is undefined?

`k1`, `k2` and `ctx` should be fields, initialized here.

Reply:
see Figure 2.2 in RFC 4493.